### PR TITLE
chore: enable telemetry reporting COMPASS-5225

### PR DESCRIPTION
**Pull request by addaleax**
### Checklist
- [ ] New tests and/or benchmarks are included
- [ ] Documentation is changed or added

## Motivation and Context

- [ ] Bugfix
- [ ] New feature
- [ ] Dependency update
- [x] Misc

## Types of changes

- [ ] *Backport Needed*
- [x] Patch (non-breaking change which fixes an issue)
- [ ] Minor (non-breaking change which adds functionality)
- [ ] Major (fix or feature that would cause existing functionality to change)

### CODE REVIEW

File: packages/compass/src/main/telemetry.ts [hunk 1]
```diff
@@ -17,7 +17,7 @@ const SEGMENT_API_KEY =
 const SEGMENT_HOST =
   process.env.HADRON_METRICS_SEGMENT_HOST_OVERRIDE ||
   process.env.HADRON_METRICS_SEGMENT_HOST;
-const IS_CI = process.env.IS_CI;
+const IS_CI = process.env.IS_CI || process.env.CI || process.env.EVERGREEN_BUILD_VARIANT;
```
Comment (addaleax) [inline packages/compass/src/main/telemetry.ts:20]: This is just me being a bit more careful, I think we have `IS_CI` already in evergreen but this way we can be sure that this is not reporting telemetry for our e2e tests :)

File: .evergreen.yml [hunk 2]
```diff
@@ -85,6 +85,7 @@ functions:
           export HADRON_METRICS_BUGSNAG_KEY="${metrics_bugsnag_key}";
           export HADRON_METRICS_INTERCOM_APP_ID="${metrics_intercom_app_id}";
           export HADRON_METRICS_STITCH_APP_ID="${metrics_stitch_app_id}";
+          export HADRON_METRICS_SEGMENT_API_KEY="${metrics_segment_api_key}";
```
Comment (mcasimir) [inline .evergreen.yml:88]: Should we continue to use `HADRON_` as a name?

Comment (addaleax) [reply .evergreen.yml:88]: I know you already brought this up in the ticket – I don’t think it matters much, I agree that it’s not really ideal, but I’d like to keep this change focused on really just enabling telemetry. We can change the prefix to `COMPASS` for all of these variables as a normal code cleanup thing.

Comment (mcasimir) [reply .evergreen.yml:88]: Ah you mean we rename everything? I'm totally fine keeping it for now in any case, my point was that the other variables will just go away with the old telemetry so there was no reason to keep it consistent

Comment (addaleax) [reply .evergreen.yml:88]: Only the stitch one will go away, though – so yeah, I’d either rename everything or nothing :)

**Review by gribnoysup — APPROVED** [review]

**Pull request merged**